sign/cosi/cosi_test.go

```diff
@@ -28,15 +28,31 @@ func (m *cosiSuite) RandomStream() cipher.Stream { return m.r }
 var testSuite = &cosiSuite{edwards25519.NewBlakeSHA256Ed25519(), blake2xb.New(nil)}
 
 func TestCoSi(t *testing.T) {
-	testCoSi(t, 2, 0)
-	testCoSi(t, 5, 0)
-	testCoSi(t, 5, 2)
-	testCoSi(t, 5, 4)
+	message := []byte("Hello World Cosi")
+	testCoSi(t, 2, 0, message)
+	testCoSi(t, 2, 0, message)
+	testCoSi(t, 5, 0, message)
+	testCoSi(t, 5, 2, message)
+	testCoSi(t, 5, 4, message)
 }
 
-func testCoSi(t *testing.T, n, f int) {
-	message := []byte("Hello World Cosi")
+func FuzzCoSi(f *testing.F) {
+	f.Fuzz(func(t *testing.T, n, f int, msg []byte) {
+		if (len(msg) < 1) || (len(msg) > 1000) {
+			t.Skip("msg must have byte length between 1 and 1000")
+		}
+		if n < 1 || n > 100 {
+			t.Skip("n must be between 1 and 100")
+		}
+		if f < 0 || f >= n {
+			t.Skip("f must be between 0 and n-1")
+		}
+
+		testCoSi(t, n, f, msg)
+	})
+}
 
+func testCoSi(t *testing.T, n, f int, message []byte) {
 	// Generate key pairs
```

Comment:
key generation probably needs to happen prior to fuzzing here too

Reply:
Key generation uses the value of n though, which is different for every fuzzer

Reply:
Actually having a random keypair in the fuzz tests is an issue, see other comments. We need either:
- an hardcoded keypair for fuzzing
- a seeded, deterministic randomness stream for fuzz tests.
-